modules/context/context.go

```diff
@@ -208,6 +209,23 @@ func (ctx *Context) HTML(status int, name base.TplName) {
 	}
 }
 
+// RenderSitemapIndex renders a sitemap pagination index from the specified baseurl
+func (ctx *Context) RenderSitemapPagination(base string, total, pagination int) {
+	idx := sitemap.NewSitemapIndex()
+	count := 1
+	for i := 0; i < total; i += pagination {
+		idx.Add(&sitemap.URL{
+			Loc: base + "-" + strconv.Itoa(count) + ".xml",
+		})
+		count++
+	}
+	ctx.Resp.WriteHeader(http.StatusOK)
+	ctx.Resp.Header().Set("Content-Type", "text/xml")
```

Comment:
```suggestion
	ctx.Resp.Header().Set("Content-Type", "text/xml")
```
That's the wrong order. After `WriteHeader` you can't change the header. The first `Write` calls `WriteHeader` with `StatusOK`,